prov/efa/src/rdm/efa_rdm_ep_fiops.c

```diff
@@ -27,30 +66,34 @@ static
 int efa_rdm_ep_create_base_ep_ibv_qp(struct efa_rdm_ep *ep)
 {
 	struct ibv_qp_init_attr_ex attr_ex = { 0 };
+	struct efa_rdm_cq *tx_rdm_cq, *rx_rdm_cq;
+	struct ibv_cq_ex *tx_ibv_cq, *rx_ibv_cq;
 
-	attr_ex.cap.max_send_wr = ep->base_ep.domain->device->rdm_info->tx_attr->size;
-	attr_ex.cap.max_send_sge = ep->base_ep.domain->device->rdm_info->tx_attr->iov_limit;
-	attr_ex.send_cq = ibv_cq_ex_to_cq(ep->ibv_cq_ex);
+	tx_rdm_cq = efa_rdm_ep_get_tx_rdm_cq(ep);
+	rx_rdm_cq = efa_rdm_ep_get_rx_rdm_cq(ep);
 
-	attr_ex.cap.max_recv_wr = ep->base_ep.domain->device->rdm_info->rx_attr->size;
-	attr_ex.cap.max_recv_sge = ep->base_ep.domain->device->rdm_info->rx_attr->iov_limit;
-	attr_ex.recv_cq = ibv_cq_ex_to_cq(ep->ibv_cq_ex);
+	if (!tx_rdm_cq && !rx_rdm_cq) {
```

Comment:
do we need this check? Is it possible to create an endpoint not bound to either tx or rx cq and without send or receive caps?

Reply:
You cannot create a QP without any CQ. It's not required before because we create IBV CQ as part of EP, but now since we need to bind the CQ to ep explicitly we need this check